pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -200,6 +200,7 @@ func (l *PodInterface) PlugPhase1(vmi *v1.VirtualMachineInstance, iface *v1.Inte
 		isMultiqueue := (vmi.Spec.Domain.Devices.NetworkInterfaceMultiQueue != nil) && (*vmi.Spec.Domain.Devices.NetworkInterfaceMultiQueue)
 		if isMultiqueue {
 			queueNumber, _ = api.CalculateNetworkQueueNumberAndGetCPUTopology(vmi)
+			queueNumber = api.CapTapDeviceQueueNumber(uint(queueNumber))
```

Comment:
I think that you should just cap the queue number already inside `CalculateNetworkQueueNumberAndGetCPUTopology`. Otherwise you will again have the potential mismatch on every place where you want the actual queue number. You would always have to call both functions in a row.

Reply:
Done.